internal/handlers/create_catalog.go

```diff
@@ -566,3 +570,23 @@ func computeImageUID(ref name.Reference, digest string) string {
 	fmt.Fprintf(sha, "%s:%s@%s", ref.Context().Name(), ref.Identifier(), digest)
 	return hex.EncodeToString(sha.Sum(nil))
 }
+
+// isPlatformAllowed verify if the platform of the image is allowed in the registry filter.
+func isPlatformAllowed(platform cranev1.Platform, allowedPlatforms []v1alpha1.Platform) bool {
+	// Images can contain "unknown/unknown" layers, which usually contain attestations.
+	// See https://docs.docker.com/build/metadata/attestations/attestation-storage/
+	// We need to skip these images, as they cannot be scanned.
+	if platform.OS == "unknown" && platform.Architecture == "unknown" {
+		return false
+	}
+
+	// If no platform is specified in the Registry CR,
+	// we assume the user wants to scan all the platforms.
+	if len(allowedPlatforms) == 0 {
+		return true
+	}
+
+	return slices.ContainsFunc(allowedPlatforms, func(p v1alpha1.Platform) bool {
+		return p.String() == platform.String()
+	})
```

Comment:
This has to be changed, I think it would not work (as in consider the plaform not allowed) these cases:

- user specified `linux/arm64/v8`, but the manifest has `linux/arm64`
- user specified `linux/arm` because he wants all the arm 32 bit variants to be scanned, but the manifest has `linux/arm/v[6,7,8]`